kart/diff_util.py

```diff
@@ -182,9 +184,19 @@ def get_file_diff(
     # We don't yet support attachment diffs in the workdir
     assert not include_wc_diff
 
-    old_tree = base_rs.tree
-    new_tree = target_rs.tree
-    repo = target_rs.repo
+    if hasattr(base_rs, "type") and hasattr(target_rs, "type"):
```

Comment:
I'd suggest changing the method signature above so it always takes trees and a repo, rather than RepoStructure objects. It's only called from a couple of places, so this won't be a big deal for callers